src/App/component/PullRequestsPage.jsx

```diff
@@ -0,0 +1,170 @@
+import moment from 'moment'
+import { Backdrop, CircularProgress } from '@material-ui/core'
+import { connect } from 'react-redux';
+
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    '& > *': {
+      margin: theme.spacing(1),
+    },
+    minWidth: '30px',
+  },
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#fff',
+  },
+}))
+
+function PullRequestsPage(prop) {
+  const classes = useStyles()
+  const [pullRequestListData, setPullRequestListData] = useState([])
+  const [dataForPullRequestChart, setDataForPullRequestChart] = useState({ labels: [], data: { opened: [], closed: [], merged: [] } })
+
+  const [currentProject, setCurrentProject] = useState({})
+
+  const projectId = localStorage.getItem("projectId")
+  const jwtToken = localStorage.getItem("jwtToken")
+
+  const [open, setOpen] = useState(false);
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const handleToggle = () => {
+    setOpen(!open);
+  };
+
+  useEffect(() => {
+    Axios.get(`http://localhost:9100/pvs-api/project/1/${projectId}`,
+      { headers: { "Authorization": `${jwtToken}` } })
+      .then((response) => {
+        setCurrentProject(response.data)
+      })
+      .catch((e) => {
+        alert(e.response?.status)
+        console.error(e)
+      })
+  }, [])
+
+  const getPullRequestsFromGitHub = () => {
+    const githubRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'github')
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+
+      // todo need reafctor with async
+      Axios.get(`http://localhost:9100/pvs-api/github/pullRequests/${query}`,
+        { headers: { "Authorization": `${jwtToken}` } })
+        .then((response) => {
+          if (response?.data) {
+            setPullRequestListData(response.data)
+          }
+        })
+        .catch((e) => {
+          alert(e.response?.status);
+          console.error(e)
+        })
+    }
+  }
+
+  useEffect(() => {
+    if (Object.keys(currentProject).length !== 0) {
+      handleToggle()
+      getPullRequestsFromGitHub()
+      handleClose()
+    }
+  }, [currentProject, prop.startMonth, prop.endMonth])
+
+  // Generate the pull-request chart
+  useEffect(() => {
+    const { startMonth, endMonth } = prop
+    let chartDataset = { labels: [], data: { merged: [], closed: [], opened: [] } }
+    let pullRequestListDataSortedByCreatedAt = pullRequestListData
+    let pullRequestListDataSortedByClosedAt = pullRequestListData
+    let pullRequestListDataSortedByMergedAt = pullRequestListData
+
+    // Sort data by date
+    if (pullRequestListData !== undefined) {
+      [].slice.call(pullRequestListDataSortedByCreatedAt).sort((a, b) => a.createdAt - b.createdAt);
+      [].slice.call(pullRequestListDataSortedByClosedAt).sort((a, b) => a.closedAt - b.closedAt);
+      [].slice.call(pullRequestListDataSortedByMergedAt).sort((a, b) => a.mergedAt - b.mergedAt);
+    }

```

Comment:
1. Suggest not let `Data` be concatenated after `List` in a variable name.

2. Suggest always use `const` to define a variable in every place, which can reduce the browser memory usage after compiling our JS code to web assembly.

3. And also suggest extract the sort process into isolated function, which may looks like:
```js
const getPRListSortedBy = (prList, key) => prList.sort((prev, curr) => prev[key] - curr[key])

const prListSortedByCreatedAt = getPRListSortedBy(prList, 'createdAt')
const prListSortedByClosedAt = getPRListSortedBy(prList, 'ClosedAt')
const prListSortedByMergedAt = getPRListSortedBy(prList, 'MergedAt')
```